### PR TITLE
Use 'git describe' to generate version number for dev version

### PR DESCRIPTION
**Pull request by queengooborg**
This PR updates the app to use `git describe` to create a version number for the development version.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -35,14 +35,23 @@ import parseResults from './results.js';
...
-  process.env.NODE_ENV === 'production' ?
-    (await fs.readJson(new URL('./package.json', import.meta.url))).version :
-    'Dev';
+let appVersion;
+if (process.env.NODE_ENV === 'production') {
+  appVersion = (await fs.readJson(new URL('./package.json', import.meta.url)))
+    .version;
+} else {
+  try {
+    appVersion = 'Dev-' + String(exec('git describe --tags')).substr(1);
```
Comment (foolip) [inline app.js:49]: The `substr(1)` bit is assuming that that the tag starts with "v" and will result in something broken if that's not the case. How about we instead prepend a "v" to the version read from package.json, and make no assumptions about what the tag found by `git describe --tags` is?

See next comment for what to do about "Dev".

File: exporter.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ const createBody = (meta) => {
     }` +
     `\nHash Digest: ${meta.digest}` +
     `\nTest URLs: ${meta.urls.join(', ')}` +
-    (meta.version == 'Dev' ?
+    (meta.version.startsWith('Dev') ?
```
Comment (foolip) [inline exporter.js:76]:
```suggestion
    (meta.version.includes('-') ?
```

File: app.js [hunk 3]
```diff
@@ -35,14 +35,23 @@ import parseResults from './results.js';
...
+  } catch (e) {
+    // If anything happens, i.e. git isn't installed, ignore
```
Comment (foolip) [inline app.js:51]: I think let's instead fall back to package.json. So if `NODE_ENV !== 'production'` we try using `git describe --tags`, and if we don't get and return a version, we read it from package.json.

**Review by foolip — CHANGES_REQUESTED** [review]

File: exporter.js (hunk 2 above)
Comment (queengooborg) [reply exporter.js:76]: I'm skeptical about using a hyphen to indicate a dev version or not, but I'm down for using `includes()` instead!

**Comment by foolip** [comment]
Hmm, I made the tests fail. Not sure why.

**Comment by queengooborg** [comment]
Ah, the tests still use just "Dev" as the version in the dummy results, so that'll need to be tweaked.

**Pull request merged**